dfastmi/io/DFastMIConfigParser.py

```diff
@@ -0,0 +1,269 @@
+
+
+class DFastMIConfigParser:
+    """
+    Class to process and parse both river and DFast MI application config files
+    """
+    _config: configparser.ConfigParser
+    """ 
+        The dictionary containing river data.
+    """
+    
+    _processor : RiverConfigElementProcessor
+    """
+        Element processer object containing processing functions and parser methods per data type
+        to retrieve River elements using DFast MI DTO Reach and branch from river config (files)
+    """
+
+    _config_processor : ConfigProcessor
+    """
+        Element processer object containing parser methods per data type
+        to retrieve DFast MI key value from DFast MI config (files)
+    """
+    
+
+    def __init__(self, config: configparser.ConfigParser):
+        self._config = config
+        self._processor = RiverConfigElementProcessor()
+        self._processor.register_processor(bool, self._process_entry_value, self._parse_bool)
+        self._processor.register_processor(int, self._process_entry_value, self._parse_int)
+        self._processor.register_processor(float, self._process_entry_value, self._parse_float)
+        self._processor.register_processor(Tuple[float, ...], self._process_tuple_entry_value, self._parse_float)
+        self._config_processor = ConfigProcessor()
+        self._config_processor.register_parser(bool, self._parse_bool)
+        self._config_processor.register_parser(int, self._parse_int)
+        self._config_processor.register_parser(float, self._parse_float)
+        self._config_processor.register_parser(str, lambda input: [input])
+
+    def _read_value(self, key: str, branch_name: str, reach_index: int):
+        """
+        reads the value from the general section, branch section or the reach section key
+        """
+        try:
+            general_value = self._config["General"][key]
+        except:
+            general_value = ''
+
+        try:
+            branch_val = self._config[branch_name][key]
+        except:
+            branch_val = general_value
+
+        try:
+            val = self._config[branch_name][key + str(reach_index)]
+        except:
+            val = branch_val
+        return val
+
+    # default processor
+    def _process_entry_value(self, key, entry_value: str, reach : Reach, parse: Callable[[str], Tuple[T, ...]], default: Optional[T], expected_number_of_values : Optional[int]) -> T:
+        if entry_value == "" and default is not None:
+            value_from_config = default
+        else:
+            try:
+                vals = parse(entry_value)
+            except:
+                vals = ()
+
+            if expected_number_of_values is None:
+                expected_number_of_values = 1
+
+            if len(vals) != expected_number_of_values:
+                self._raise_exception_incorrect_value_entries(key, reach.name, reach.parent_branch.name, entry_value, expected_number_of_values)
+            value_from_config = vals[0]
```

Comment:
if it should always be expected_number_of_values of 1, this can just be vals right? since there is only 1.